crates/sui-graphql-rpc/src/types/move_module.rs

```diff
@@ -46,12 +48,111 @@ impl MoveModule {
         }
     }
 
-    // friendConnection(
-    //   first: Int,
-    //   after: String,
-    //   last: Int,
-    //   before: String
-    // ): MoveModuleConnection
+    /// Modules that this module considers friends (these modules can access `public(friend)`
+    /// functions from this module).
+    async fn friend_connection(
+        &self,
+        ctx: &Context<'_>,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Connection<String, MoveModule>> {
+        // TODO: make cursor opaque (currently just an offset).
+        validate_cursor_pagination(&first, &after, &last, &before).extend()?;
+
+        let bytecode = self.parsed.bytecode();
+        let total = bytecode.friend_decls.len();
+
+        // Add one to make [lo, hi) a half-open interval ((after, before) is an open interval).
```

Comment:
This comment is a bit confusing me. How would the user know that this happens under the hood, just by using `after` variable name, we expect that we get after + 1?

Reply:
`after` and `before` are both bounds on an open interval (exclusive bounds) -- this is specified as part of the [Relay Connection Spec](https://relay.dev/graphql/connections.htm#sec-Pagination-algorithm) -- I'm converting the interval into a half open one (where the lower bound is inclusive) because that is what is most convenient for indexing.  So, this isn't something that the user would be aware of at all.